### ISSUE TITLE
Unexpected behaviour when using min with NSE

### ISSUE THREAD

**Issue by dirkschumacher** [tidyverse/dplyr#3857]
It seems that `min` is treated differently compared to other functions in `mutate` statements when used with NSE. I created a small rexprex that illustrates what I mean.  

The agrument to `nse` in `call1` is already evaluated, while in `call2` `expr` still holds the unevaluated code `min2(cyl)`.
```r
nse <- function(expr) {
  print(substitute(expr))
  1
}
min2 <- min
tmp <- dplyr::mutate(mtcars, call1 = nse(min(cyl)), call2 = nse(min2(cyl)))
#> [1] 4
#> min2(cyl)
```

I am not sure if this is a bug or feature. But I have a usecase, where I would like to access the quoted code of the argument as well as the evaluated value.

**Comment by romainfrancois** [comment]
This has been fixed in #3799. I get this with the current master: 

``` r
nse <- function(expr) {
  print(substitute(expr))
  1
}
min2 <- min
call1 <- quote(nse(min(cyl)))
call2 <- quote(nse(min2(cyl)))
tmp <- dplyr::mutate(mtcars, !!call1, !!call2)
#> min(cyl)
#> min2(cyl)
rlang::qq_show(dplyr::mutate(mtcars, !!call1, !!call2))
#> dplyr::mutate(mtcars, nse(min(cyl)), nse(min2(cyl)))
```

What used to happen is that the old hybrid evaluator would recognize `min(cyl)` and eagerly replace that with its result. 

What we have now is less surprising behavior where hybrid only kicks in if it handles the entire expression.

**Comment by dirkschumacher** [comment]
Ah perfect. Thanks👍 Looking forward to the new release ;)
